### PR TITLE
DeploymentManager: Fix property name and enhance descriptions

### PR DESCRIPTION
**Pull request by deveshguha**
The changes include updates to property descriptions that adds more context and clarity. 
There is one change to the contract - update of a property name 'dependsOnStepGroup' to 'dependsOnStepGroups'. The service is in preview with a sign-up process and there are no allowed users at this time, hence the API version is not updated. 

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/deploymentmanager/resource-manager/readme.nodejs.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ Please also specify `--node-sdks-folder=<path to root folder of your azure-sdk-f
 nodejs:
   azure-arm: true
   package-name: azure-arm-deploymentmanager
+  package-version: 0.9.0
```
Comment (daschult) [inline specification/deploymentmanager/resource-manager/readme.nodejs.md]: Please remove this property. We no longer put package-versions in these readme files.

**Comment by annatisch** [comment]
Hi @deveshguha - are you aware that this is a breaking change that will require a major version bump for any stable published SDKs?

If you have not yet published SDKs, or they are only in preview, this shouldn't be an issue - just making sure you are aware of that :)

**Comment by deveshguha** [comment]
> Hi @deveshguha - are you aware that this is a breaking change that will require a major version bump for any stable published SDKs?
> 
> If you have not yet published SDKs, or they are only in preview, this shouldn't be an issue - just making sure you are aware of that :)

@annatisch, thank you for clarifying how to handle published SDKs and we do not have any :) Yes, I am aware this is a breaking change and I've captured the rationale for not bumping up the version in the PR description.

**Review by hovsepm — APPROVED** [review]
All clear from my side....

**Review by annatisch — APPROVED** [review]
LGTM

**Pull request merged**